cmd/auth-rpc-client.go

```diff
@@ -146,20 +146,26 @@ func (authClient *AuthRPCClient) Call(serviceMethod string, args interface {
 	SetToken(token string)
 	SetTimestamp(tstamp time.Time)
 }, reply interface{}) (err error) {
-	// On successful login, attempt the call.
-	if err = authClient.Login(); err == nil {
+	loginAndCallFn := func() error {
+		// On successful login, proceed to attempt the requested service method.
+		if err = authClient.Login(); err != nil {
+			return err
+		}
 		// Set token and timestamp before the rpc call.
 		args.SetToken(authClient.serverToken)
 		args.SetTimestamp(time.Now().UTC())
 
-		// Call the underlying rpc.
-		err = authClient.rpc.Call(serviceMethod, args, reply)
-
-		// Invalidate token, and mark it for re-login on subsequent reconnect.
+		// Finally make the network call using net/rpc client.
+		return authClient.rpc.Call(serviceMethod, args, reply)
+	}
+	if err = loginAndCallFn(); err != nil {
+		// Invalidate token, and mark it for re-login and reconnect.
 		if err == rpc.ErrShutdown {
-			authClient.mu.Lock()
-			authClient.isLoggedIn = false
-			authClient.mu.Unlock()
+			// Close the underlying connection.
+			authClient.Close()
+
+			// Call login and call again.
+			err = loginAndCallFn()
```

Comment:
This logic makes one more attempt to execute on `ErrShutdown`.  It would be nice to have a logic to make multiple attempts if its wanted in future meaning just changing a constant will do more or less attempts.  In short, you could have a `for` loop.